src/mazegeneration/PrimsGenerator.tsx

```diff
@@ -0,0 +1,53 @@
+
+import { BoardEnum } from "../utils/BoardConfig";
+
+interface Point {
+  row: number;
+  column: number;
+}
+
+export function generateMazePrims(rows: number, columns: number, startIdx: Point, endIdx: Point) {
+  let grid: number[][] = Array(rows).fill(null).map(() => Array(columns).fill(BoardEnum.WALL));
+
+  let toVisit: Point[] = [];
+  toVisit.push(startIdx);
+
+  let dx = [0, 1, 0, -1];
+  let dy = [1, 0, -1, 0];
+
+  while (toVisit.length > 0) {
+    let randomIndex = Math.floor(Math.random() * toVisit.length);
+    let point = toVisit[randomIndex];
+    toVisit.splice(randomIndex, 1);
+
+    let neighbors: Point[] = [];
+    for (let i = 0; i < 4; i++) {
+      let newRow = point.row + 2 * dx[i];
+      let newColumn = point.column + 2 * dy[i];
+      if (
+        newRow >= 0 &&
```

Comment:
Throw this in a function, if its the same as the check you made in the bfs file, then just import that over

Reply:
Not the same check, isValid in BFS has visited array and it also does the opposite check. Instead of !== BoardEnum.WALL it does ===BoardEnum.WALL

Reply:
I'm sorry about the confusion, here's an article to better familiarize yourself with an implication:
https://math.libretexts.org/Bookshelves/Combinatorics_and_Discrete_Mathematics/A_Spiral_Workbook_for_Discrete_Mathematics_(Kwong)/02%3A_Logic/2.03%3A_Implications#:~:text=An%20implication%20is%20the%20compound,true%20in%20all%20other%20situations.

Happy coding!

Reply:
No worries, thanks for clearing it up!

https://www.canada.ca/en/public-health/news/2023/03/message-from-the-minister-of-health--world-obesity-day-2023.html

Hope you find your toes!